### PR TITLE
Added Perfect Square String

### LINKED ISSUE #1280: Perfect Square String
in c, c++, Java, python and go

### PR DESCRIPTION
**Pull request by somya-kapoor**
Fixes #1280

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Added Perfect Square String in c, c++, Java, python and go

#### Languages Used:
- C
- C++
- java 
- python
- go

#### Files Added:
- Perfect_Square_String.c 
- Perfect_Square_String.cpp
- Perfect_Square_String.go
- Perfect_Square_String.java
- Perfect_Square_String.py

### CODE REVIEW

File: Perfect_Square_String/Perfect_Square_String.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# Driver code 
+str = input(); 
+  
+if (PerfectSquareString(str) == 0): 
+    print("YES"); 
+else: 
+    print("NO");
+    
+'''
+/*
```
Comment (jainaman224) [inline Perfect_Square_String/Perfect_Square_String.py]: Typo

File: Perfect_Square_String/Perfect_Square_String.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+INPUT:
```
Comment (jainaman224) [inline Perfect_Square_String/Perfect_Square_String.py]: Indent this.

File: Perfect_Square_String/Perfect_Square_String.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+        if (PerfectSquareString(str) == 0) 
+            System.out.println("YES"); 
+        else
+            System.out.println("NO"); 
+    } 
+} 
+
+/*
+INPUT:
```
Comment (jainaman224) [inline Perfect_Square_String/Perfect_Square_String.java]: Indent this.

**Comment by somya-kapoor** [comment]
@jainaman224 Done please review

**Comment by somya-kapoor** [comment]
@jainaman224 Done, Please review.

File: Perfect_Square_String/Perfect_Square_String.c [hunk 4]
```diff
@@ -0,0 +1,43 @@
+/*Perfect Square String is a string in which sum of ASCII value of all characters is a perfect square.*/ 
+
+#include <stdio.h> 
+#include <string.h>
+#include <math.h>
```
Comment (MastersAbh) [inline Perfect_Square_String/Perfect_Square_String.c]: remove the spaces after include

File: Perfect_Square_String/Perfect_Square_String.cpp [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+// Driver code 
+int main() 
+{ 
+    string str;
+    cin >> str;
+      
+    if (perfectSquareString(str) == 0) 
+        cout << "YES"; 
+    else
+        cout << "NO";
```
Comment (MastersAbh) [inline Perfect_Square_String/Perfect_Square_String.cpp]: remove spaces within cin and cout statements.

File: Perfect_Square_String/Perfect_Square_String.go [hunk 6]
```diff
@@ -0,0 +1,44 @@
+/*Perfect Square String is a string in which sum of ASCII value of all characters is a perfect square.*/ 
+
+package main
+import (
```
Comment (MastersAbh) [inline Perfect_Square_String/Perfect_Square_String.go]: a blank line before import

File: Perfect_Square_String/Perfect_Square_String.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
+'''
+Perfect Square String is a string in which sum of ASCII value of all characters is a perfect square.
+'''
+
+import math;
```
Comment (MastersAbh) [inline Perfect_Square_String/Perfect_Square_String.py]: please remove semicolon. It isn't used in python

File: Perfect_Square_String/Perfect_Square_String.py [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        sum = sum + ord(str[i]); 
+      
+    # Calculating square root 
+    squareRoot = math.sqrt(sum); 
+      
+    # Checking if square root is an integer 
+    return squareRoot - math.floor(squareRoot)); 
+  
+# Driver code 
+str = input();
```
Comment (MastersAbh) [inline Perfect_Square_String/Perfect_Square_String.py]: str is a keyword in python. A better variable name would be nice.

**Comment by somya-kapoor** [comment]
@MastersAbh Done please review

File: Perfect_Square_String/Perfect_Square_String.cpp [hunk 9]
```diff
@@ -0,0 +1,42 @@
+/*Perfect Square String is a string in which sum of ASCII value of all characters is a perfect square.*/ 
+
+#include <bits/stdc++.h>
```
Comment (MastersAbh) [inline Perfect_Square_String/Perfect_Square_String.cpp]: remove space before <

File: Perfect_Square_String/Perfect_Square_String.go [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    "fmt"
+    "math"
+)
+
+func perfectSquareString(str string) float64 { 
+    var sum float64 = 0 
+      
+    // Length of the string 
+    var length int = len(str);
```
Comment (MastersAbh) [inline Perfect_Square_String/Perfect_Square_String.go]: remove semicolons. not required in Go

File: Perfect_Square_String/Perfect_Square_String.py [hunk 11]
```diff
@@ -0,0 +1,36 @@
+'''
+Perfect Square String is a string in which sum of ASCII value of all characters is a perfect square.
+'''
+
+import math
+def perfectSquareString(string):
```
Comment (MastersAbh) [inline Perfect_Square_String/Perfect_Square_String.py]: A blank line before function declaration

File: Perfect_Square_String/Perfect_Square_String.py [hunk 12]
```diff
@@ -0,0 +1,36 @@
+'''
+Perfect Square String is a string in which sum of ASCII value of all characters is a perfect square.
+'''
+
+import math
+def perfectSquareString(string): 
+    sum = 0;
```
Comment (MastersAbh) [inline Perfect_Square_String/Perfect_Square_String.py]: semicolons not required in python

**Comment by somya-kapoor** [comment]
@MastersAbh Done, Please review.

**Comment by somya-kapoor** [comment]
@jainaman224  Please review

File: Perfect_Square_String/Perfect_Square_String.py [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+        sum = sum + ord(string[i])
+      
+    # Calculating square root 
+    squareRoot = math.sqrt(sum) 
+      
+    # Checking if square root is an integer 
+    return squareRoot - math.floor(squareRoot))
+  
+# Driver code 
+string = input();
```
Comment (MastersAbh) [inline Perfect_Square_String/Perfect_Square_String.py]: please dont add semicolon in python files

**Comment by somya-kapoor** [comment]
@MastersAbh Done, Please review.

**Comment by somya-kapoor** [comment]
@jainaman224 Please review

**Pull request merged**